packages/venia-concept/src/components/Checkout/braintreeDropin.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * @fileoverview This component uses the BrainTree Web Drop In to
+ * hook into Web Payments and the Payment Request API to
+ * submit payments via BrainTree.
+ *
+ * See:
+ * https://github.com/braintree/braintree-web-drop-in
+ * https://braintree.github.io/braintree-web-drop-in/docs/current/index.html
+ * https://developers.braintreepayments.com/guides/drop-in/overview/javascript/v3
+ */
+
+import React, { Component } from 'react';
+import { bool, func } from 'prop-types';
+import dropin from 'braintree-web-drop-in';
+
+const { BRAINTREE_TOKEN } = process.env;
+
+class BraintreeDropin extends Component {
+    static propTypes = {
+        isRequestingPaymentNonce: bool,
+        onError: func.isRequired,
+        onSuccess: func.isRequired
+    };
+
+    async componentDidMount() {
+        this.braintree = await this.createDropinInstance();
+    }
+
+    componentDidUpdate(prevProps) {
+        if (
+            this.braintree &&
+            this.props.isRequestingPaymentNonce &&
+            !prevProps.isRequestingPaymentNonce
+        ) {
+            // Our parent is telling us to request the payment nonce.
+            this.requestPaymentNonce();
+        }
+    }
+
+    render() {
+        return <div id="dropin-container" />;
+    }
+
+    createDropinInstance = () => {
+        // Create an instance of the BrainTree Web Drop In.
+        return dropin.create({
+            authorization: BRAINTREE_TOKEN,
+            // Note: this selector must match to some part of this component's rendered HTML.
+            container: '#dropin-container',
+            card: {
+                // Show a cardholder name field.
+                cardholderName: true,
+                overrides: {
+                    fields: {
+                        number: {
+                            maskInput: {
+                                // Only show the last four digits of the credit card number after focus exits this field.
+                                showLastFour: true
+                            }
+                        }
+                    }
+                }
+            }
+        });
+    };
+
+    /*
+     * @see https://developers.braintreepayments.com/guides/payment-method-nonces.
+     * @see https://braintree.github.io/braintree-web-drop-in/docs/current/Dropin.html#requestPaymentMethod.
+     */
+    requestPaymentNonce = async () => {
+        try {
+            const paymentNonce = await this.braintree.requestPaymentMethod();
+            this.props.onSuccess(paymentNonce);
+        } catch (e) {
+            // BrainTree will update the UI with error messaging,
+            // but signal that there was an error.
+            console.warn(`Invalid Payment Details. \n${e}`);
+            this.props.onError();
+        }
+    };
+}
+
+export default BraintreeDropin;
```

Comment:
Is this component not wrapped with `classify`? Why don't we put any classes on the root `div` here?

Reply:
Fixed with update 👍